# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContextGroup.h
# Commit: 81c6cd7f8b5b
# Full Hash: 81c6cd7f8b5bd2750c00cc092d67199092b4ed87
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2025-03-04 04:58:00
# Regressor Bug: 1665474
# File Overlap Count: 1
# Description:
#   Bug 1665474 - Part 9: Handle calls to browser.createAboutBlankDocumentViewer with http(s) principals, r=farre
#   
#   It is possible for frontend JS to call this method on a browser element
#   with an http(s) principal. In this case, treat the new about:blank
#   document like it was created without a Origin-Agent-Cluster header, and
# ==============================================================================

diff -r 5a4d6c2e0dec -r 81c6cd7f8b5b docshell/base/BrowsingContextGroup.h
--- a/docshell/base/BrowsingContextGroup.h	Mon Mar 03 22:51:30 2025 +0000
+++ b/docshell/base/BrowsingContextGroup.h	Mon Mar 03 22:51:30 2025 +0000
@@ -225,18 +225,20 @@
   // process isolation.
   // The relevant subset of this mapping is mirrored to content processes, to be
   // used when determining DocGroup keying.
-  // UsesOriginAgentCluster returns `Nothing()` if whether or not to origin
-  // isolate a given principal is unknown. This should never happen in the
-  // content process, as all navigations resulting in http(s) document loads
-  // should originate in the parent process, or occur within a
-  // BrowsingContextGroup which has already loaded documents with the given
-  // principal.
   void SetUseOriginAgentClusterFromNetwork(nsIPrincipal* aPrincipal,
                                            bool aUseOriginAgentCluster);
   void SetUseOriginAgentClusterFromIPC(nsIPrincipal* aPrincipal,
                                        bool aUseOriginAgentCluster);
   Maybe<bool> UsesOriginAgentCluster(nsIPrincipal* aPrincipal);
 
+  // Ensures that the given principal will return `Some(...)` from
+  // `UsesOriginAgentCluster` going forward, setting it to a default value if no
+  // value is set.
+  // As the map can only be modified in the parent process, this method will
+  // crash if there is no CrossOriginIsolatedStatus for the principal when
+  // called in a content process.
+  void EnsureUsesOriginAgentClusterInitialized(nsIPrincipal* aPrincipal);
+
   void ChildDestroy();
 
  private: